src/transformers/models/realm/retrieval_realm.py

```diff
@@ -0,0 +1,160 @@
+# coding=utf-8
+# Copyright 2021 The REALM authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""REALM Retriever model implementation."""
+
+import os
+from typing import Optional, Union
+
+import numpy as np
+
+from huggingface_hub import hf_hub_download
+
+from ...utils import logging
+from .tokenization_realm import RealmTokenizer
+
+
+_REALM_BLOCK_RECORDS_FILENAME = "block_records.npy"
+
+
+logger = logging.get_logger(__name__)
+
+
+def convert_tfrecord_to_np(block_records_path, num_block_records):
```

Comment:
is this function used anywhere?

Reply:
No good point though! I thought we could leave it as it's necessary to see how to convert between the original records on the "new" numpy methods - but would also be ok with deleting it

Reply:
I use this function in the checkpoint conversion [script](https://github.com/qqaatw/pytorch-realm-orqa/blob/abe122472aa50b71289eab77f90fc7935553e5a6/model.py#L10). 

As Patrick said, it shows how tf records are converted, and deleting it would be ok too.